digdag-standards/src/main/java/io/digdag/standards/operator/MailOperatorFactory.java

```diff
@@ -288,12 +311,16 @@ private InternetAddress newAddress(String str)
         }
         SmtpConfig config = ImmutableSmtpConfig.builder()
                 .host(userHost.get())
+                // TODO: This code expects `params` has `port` field even if `secrets` has the field.
+                //       Maybe we need to revisit here later to see whether this behavior is reasonable or not.
                 .port(secrets.getSecretOptional("port").transform(Integer::parseInt).or(params.get("port", int.class)))
                 .startTls(secrets.getSecretOptional("tls").transform(Boolean::parseBoolean).or(params.get("tls", boolean.class, true)))
                 .ssl(secrets.getSecretOptional("ssl").transform(Boolean::parseBoolean).or(params.get("ssl", boolean.class, false)))
                 .debug(params.get("debug", boolean.class, false))
                 .username(secrets.getSecretOptional("username").or(params.getOptional("username", String.class)))
                 .password(secrets.getSecretOptional("password"))
+                .connectTimeout(params.get(CONFIG_KEY_CONNECT_TIMEOUT, DurationParam.class, DEFAULT_CONNECT_TIMEOUT))
+                .socketTimeout(params.get(CONFIG_KEY_SOCKET_TIMEOUT, DurationParam.class, DEFAULT_SOCKET_TIMEOUT))
```

Comment:
It seems mail> support most of parameters for smtp connection in secrets.
So it may be better to follow the manner.
But I guess we don't need to store the parameters excepts `password` in secrets. 
So the current implementation is reasonable.

Reply:
Yeah, it's what I thought as well and I wanted to go with a simpler way to reduce the complexity.